### PR TITLE
[gps_i2c] gps_i2c module rewritten, added ubx ucenter compatibility

### PR DESCRIPTION
**Pull request by masierra**
unbroke gps_i2c module and added ucenter compatibility

```
<subsystem name="gps"           type="ublox">
  <configure name="UBX_GPS_PORT" value="gps_i2c"/>
</subsystem>
```

### CODE REVIEW

**Comment by flixr** [comment]
Nice! Will be great to finally have a functioning ublox i2c driver again!

I wonder though if there isn't a better way to set up and use a I2C link device...
@gautierhattenberger what do you think?

File: sw/airborne/modules/gps/gps_i2c.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2009  ENAC, Pascal Brisset, Michel Gorraz,Gautier Hattenberger
+ * Copyright (C) 2016 Michael Sierra <sierramichael.a@gmail.com>
```
Comment (gautierhattenberger) [inline sw/airborne/modules/gps/gps_i2c.c]: It is usually better to keep the previous authors, even when the original code sucks...

**Comment by gautierhattenberger** [comment]
The way UBX gps are using i2c is really stupid. They more or less do stream over i2c.
I think the complete gps_ubx driver should be reorganized to have the generic part independent of the uart/i2c part.
Also their is a lot of `ifdef` in ucenter module. Not sure if it can be improved.

I would say that have this thing working is already a good thing and we could merge it as his.
Future improvements should cover:
- reorganizing gps_ubx driver
- moving gps_i2c files to `modules/gps` folder (or we do that now)
- implementing the generic pprzlink_device for i2c in the actual i2c driver

File: sw/airborne/modules/gps/gps_i2c.c (hunk 1 above)
Comment (masierra) [inline sw/airborne/modules/gps/gps_i2c.c]: fixed

**Comment by flixr** [comment]
Also since this module works only for Ublox I2C GPSs, we should probably rename the module to `gps_ublox_i2c` and make it depend on `gps_ublox`.

Even better would if we could add the relevant source files in the normal gps_ublox module if i2c is given as port... that part would actually already be possible, but adding the extra init/periodic/event functions not... so maybe an improvement for later...

**Comment by flixr** [comment]
@masierra Since subsystem will go away (be replaced by modules), you should replace `subsystem` by `module`, in this case:
```
<module name="gps"           type="ublox">
  <configure name="UBX_GPS_PORT" value="gps_i2c"/>
</module>
```

**Comment by flixr** [comment]
Can you please also add doxygen blocks to the files that are missing them?
See http://docs.paparazziuav.org/latest/styledoxygen.html

**Pull request merged**